File: cawbs/DotNet/Result.cs
// Copyright (c) Core DF. All rights reserved.

using System.Text.Json;
using System.Text.Json.Serialization;

namespace CoreAuto.Cawbs;

public sealed class Result
{
    [JsonPropertyName("status_code")]
    public int StatusCode { get; init; }

    [JsonPropertyName("error")]
    [JsonIgnore(Condition = JsonIgnoreCondition.WhenWritingNull)]
    public object? Error { get; init; }

    [JsonPropertyName("payload")]
    [JsonIgnore(Condition = JsonIgnoreCondition.WhenWritingNull)]
    public JsonElement? Payload { get; init; }

    [JsonPropertyName("answer")]
    [JsonIgnore(Condition = JsonIgnoreCondition.WhenWritingNull)]
    public Dictionary<string, JsonElement>? Answer { get; init; }
}

File: cawbs/Scala/src/main/scala/com/coredf/cawbs/Result.scala
// Copyright (c) Core DF. All rights reserved.

package com.coredf.cawbs

final case class Result(
    statusCode: Int,
    error: Option[Any] = None,
    payload: Option[Any] = None,
    answer: Option[Map[String, Any]] = None
):
  def toMap: Map[String, Any] =
    Map("status_code" -> statusCode) ++
      error.map("error" -> _).toMap ++
      payload.map("payload" -> _).toMap ++
      answer.map("answer" -> _).toMap
